numpy/core/memmap.py

```diff
@@ -309,3 +309,20 @@ def flush(self):
         """
         if self.base is not None and hasattr(self.base, 'flush'):
             self.base.flush()
+
+    def __array_wrap__(self, arr, context=None):
+        # Return memmap if self and arr are likely to share memory
+        if np.may_share_memory(self, arr):
```

Comment:
Playing a bit around, I wasn't able to find an edge case in `__array_wrap__` where `np.may_share_memory(self, arr)` and `arr is self` disagreed, so I made the advised change.

If someone else can see an edge case that I missed, please complain !